### ISSUE TITLE
fMRIprep not starting (hanging with no errors after run command) on HCP using Singularity

### ISSUE THREAD

**Issue by bwinsto2** [nipreps/fmriprep#2510]
### What version of fMRIPrep are you using?
20.2.3
### What kind of installation are you using? Containers (Singularity, Docker), or "bare-metal"?
Singularity 3.6.0. Used `singularity build fmriprep_latest.simg docker://nipreps/fmriprep:20.2.3`
### What is the exact command-line you used?
```
singularity run --cleanenv -B /dcl01/smart/data/hpc900:/dcl01/smart/data/hpc900 fmriprep_latest.simg /dcl01/smart/data/hpc900/disk1_bids /dcl01/smart/data/hpc900/bwinston/derivatives participant --fs-license-file /dcl01/smart/data/hpc900/bwinston/license.txt --output-spaces anat --participant-label 103818
```

### Have you checked that your inputs are BIDS valid?
Yes. Actually, I used the same command on a local linux machine, and everything ran fine. 

### Are you reusing previously computed results (e.g., FreeSurfer, Anatomical derivatives, work directory of previous run)?
No.

### Issue 
Hi :) trying to run fMRIprep on a cluster. I requested 200G of memory, so that shouldn't be the issue. After I run the above `singularity run` command, there is no terminal output. fMRIprep creates a work directory; inside is `bids_db` (inside that is `layout_index.sqlite`). But that's it. Otherwise hanging. I searched the issues but wasn't able to find anyone with this problem. I'm quite perplexed: any ideas on what could be going on? 

Big thanks.

**Comment by effigies** [comment]
Thanks for the detailed report!

My first guess would be that your working directory is on a network filesystem, which can dramatically slow down the indexing process. If you set `--bids-database-dir` to a directory that is local to the compute node, that should recover the speed. It doesn't need to be a large filesystem, though if you have 200GB you could probably do it on a `tmpfs` and get peak performance.

**Comment by bwinsto2** [comment]
Thanks! I will reach out the sysadmin to see about storing files on a compute node, but this might be hard since I'm hoping to run a bunch of subjects in parallel on different nodes, and I will not always have 200GB of memory. Do you have a ballpark idea of how long indexing would take if I left things as-is? Anything on the order of hours would be fine since I just have to run this once on all the subjects, but I don't want to be waiting for weeks! 

Thanks again.

**Comment by bwinsto2** [comment]
Thanks! You were absolutely right. Hate to be that guy, but I'm now experiencing a perplexing BIDS issue that is totally unrelated. Pls let me know if I should open a separate issue. I'm getting a quick validation error as seen below:

```
[compute-066 /users/bwinston]$ singularity run --cleanenv -B /dcl01/smart/data/hpc900:/dcl01/smart/data/hpc900 fmriprep_latest.simg /dcl01/smart/data/hpc900/disk1_bids /dcl01/smart/data/hpc900/bwinston/derivatives participant --fs-license-file /dcl01/smart/data/hpc900/bwinston/license.txt --output-spaces anat --participant-label 103818
bids-validator@1.4.0

	1: [ERR] Quick validation failed - the general folder structure does not resemble a BIDS dataset. Have you chosen the right folder (with "sub-*/" subfolders)? Check for structural/naming issues and presence of at least one subject. (code: 61 - QUICK_VALIDATION_FAILED)
		.disk1_bids
```

Strangely, I tried using the cli BIDS validator on the same directory, and I got some warnings, but it was valid. 

```
[compute-066 /users/bwinston]$ singularity run -B /dcl01/smart/data/hpc900/:/dcl01/smart/data/hpc900/ bids_validator.sif /dcl01/smart/data/hpc900/disk1_bids
bids-validator@1.8.1-dev.0

	1: [WARN] Not all subjects contain the same files. Each subject should contain the same number of files with the same naming unless some files are known to be missing. (code: 38 - INCONSISTENT_SUBJECTS)
		./sub-100206/fmap/sub-100206_run-03_magnitude.nii.gz
			Evidence: Subject: sub-100206; Missing file: sub-100206_run-03_magnitude.nii.gz
		./sub-100206/fmap/sub-100206_run-03_phasediff.json
			Evidence: Subject: sub-100206; Missing file: sub-100206_run-03_phasediff.json
		./sub-100206/fmap/sub-100206_run-03_phasediff.nii.gz
			Evidence: Subject: sub-100206; Missing file: sub-100206_run-03_phasediff.nii.gz
		./sub-100206/fmap/sub-100206_run-04_magnitude.nii.gz
			Evidence: Subject: sub-100206; Missing file: sub-100206_run-04_magnitude.nii.gz
		./sub-100206/fmap/sub-100206_run-04_phasediff.json
			Evidence: Subject: sub-100206; Missing file: sub-100206_run-04_phasediff.json
		./sub-100206/fmap/sub-100206_run-04_phasediff.nii.gz
			Evidence: Subject: sub-100206; Missing file: sub-100206_run-04_phasediff.nii.gz
		./sub-100307/fmap/sub-100307_run-03_magnitude.nii.gz
			Evidence: Subject: sub-100307; Missing file: sub-100307_run-03_magnitude.nii.gz
		./sub-100307/fmap/sub-100307_run-03_phasediff.json
			Evidence: Subject: sub-100307; Missing file: sub-100307_run-03_phasediff.json
		./sub-100307/fmap/sub-100307_run-03_phasediff.nii.gz
			Evidence: Subject: sub-100307; Missing file: sub-100307_run-03_phasediff.nii.gz
		./sub-100307/fmap/sub-100307_run-04_magnitude.nii.gz
			Evidence: Subject: sub-100307; Missing file: sub-100307_run-04_magnitude.nii.gz
		... and 675 more files having this issue (Use --verbose to see them all).

	Please visit https://neurostars.org/search?q=INCONSISTENT_SUBJECTS for existing conversations about this issue.

	2: [WARN] Each _phasediff.nii[.gz] file should be associated with a _magnitude1.nii[.gz] file. (code: 92 - MISSING_MAGNITUDE1_FILE)
		./sub-100206/fmap/sub-100206_run-01_phasediff.nii.gz
		./sub-100206/fmap/sub-100206_run-02_phasediff.nii.gz
		./sub-100307/fmap/sub-100307_run-01_phasediff.nii.gz
		./sub-100307/fmap/sub-100307_run-02_phasediff.nii.gz
		./sub-100408/fmap/sub-100408_run-01_phasediff.nii.gz
		./sub-100408/fmap/sub-100408_run-02_phasediff.nii.gz
		./sub-100610/fmap/sub-100610_run-01_phasediff.nii.gz
		./sub-100610/fmap/sub-100610_run-02_phasediff.nii.gz
		./sub-100610/fmap/sub-100610_run-03_phasediff.nii.gz
		./sub-101006/fmap/sub-101006_run-01_phasediff.nii.gz
		... and 323 more files having this issue (Use --verbose to see them all).

	Please visit https://neurostars.org/search?q=MISSING_MAGNITUDE1_FILE for existing conversations about this issue.

	3: [WARN] The recommended file /README is missing. See Section 03 (Modality agnostic files) of the BIDS specification. (code: 101 - README_FILE_MISSING)

	Please visit https://neurostars.org/search?q=README_FILE_MISSING for existing conversations about this issue.


        Summary:                   Available Tasks:        Available Modalities: 
        8840 Files, 16.44GB        Gambling                MRI                   
        113 - Subjects                                                           
        1 - Session                                                              


	If you have any questions, please post on https://neurostars.org/tags/bids.
```
When I run fmriprep skipping bids validation, I get `fmriprep: error: One or more participant labels were not found in the BIDS directory: 103818.`

A bit confused on how to proceed. Thank you so much.

**Comment by bwinsto2** [comment]
Thanks! You were absolutely right. It turns out the files in this directory were also dangling symlinks, which caused fMRIprep not to work.
